V1_MIGRATION_GUIDE.md

```diff
@@ -0,0 +1,96 @@
+# V1 Migration Guide
+
+Guide to migrating from `v0.x` to `v1.x` of the Auth0 CLI.
+
+## Commands Reorganization
+
+Some commands have been reorganized to establish a more systematic hierarchy. All other facets of the commands (arguments, flags, etc) remain the same.
+
+| **Before (v0)**            | **After (v1)**                    |
+| -------------------------- | --------------------------------- |
+| `auth0 branding domains`   | `auth0 domains`                   |
+| `auth0 branding emails`    | `auth0 email templates`           |
+| `auth0 branding show`      | `auth0 universal-login show`      |
+| `auth0 branding update`    | `auth0 universal-login update`    |
+| `auth0 branding templates` | `auth0 universal-login templates` |
+| `auth0 branding texts`     | `auth0 universal-login prompts`   |
+
+## Authenticating With Client Credentials
+
+The `auth0 tenants add` command which enabled authenticating to a tenant via client credentials has been consolidated into the `auth0 login` command. It can be interfaced interactively through the terminal or non-interactively by passing in the client credentials with flags (example below).
+
+**Before:**
+
+```sh
+auth0 tenants add travel0.us.auth0.com \
+--client-id tUIvPH7g2ykVm4lGriYEQ6BKV3je24Ka \
+--client-secret zvKP0uWfsF0mBEgAOCgAMdFXgthfwgt_GXf9eEeMOWEIOlPK8pc3b119qBL0b2av
+```
+
+**After:**
+
+```sh
+auth0 login --domain travel0.us.auth0.com \
+--client-id tUIvPH7g2ykVm4lGriYEQ6BKV3je24Ka \
+--client-secret zvKP0uWfsF0mBEgAOCgAMdFXgthfwgt_GXf9eEeMOWEIOlPK8pc3b119qBL0b2av
+```
+
+## JSON Output Flag
+
+The `--format json` flag-value pair has been condensed into the `--json` flag.
+
+**Before:** `auth0 apps list --format json`
+
+**After:** `auth0 apps list --json`
+
+## Log Streams
+
+In `v0.x`, the creation and updating of log streams through the `auth0 logs streams create` and `auth0 log streams update` commands facilitated the management of all log stream types with a mass of type-specific flags. For `v1.x`, the type of log stream is now required as an argument. This change facilitates more ergonomic flags and type-specific validations.
+
+**Before:**
+
+```sh
+auth0 logs streams create \
+--type datadog \
+--name "My Datadog Log Stream" \
+--datadog-id us \
+--datadog-key 3c0c4965368b6b10f8640dbda46abfdc
+```
+
+**After:**
+
+```sh
+auth0 logs streams create datadog \
+--name "My Datadog Log Stream" \
+--region us \
+--api-key 3c0c4965368b6b10f8640dbda46abfdc
+```
+
+## Reveal Client Secret Flag
+
+In `v0.x`, the `auth0 apps create` command has a `--reveal` flag that would unobscure the client secrets from the output. This flag has changed to `--reveal-client-secret` for clarify what is being revealed.
```

Comment:
Isn't this `--reveal-secrets` now?

Reply:
Because it's not just the client secret that gets revealed when `--json` is passed: https://github.com/auth0/auth0-cli/blob/main/internal/display/apps.go#L244

Reply:
This got resolved in https://github.com/auth0/auth0-cli/pull/620.